pcap4j-core/src/main/java/org/pcap4j/packet/GtpV1PduSessionContainerExtensionHeader.java

```diff
@@ -0,0 +1,490 @@
+ * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * </pre>
+ *     UL PDU Session Information frames PDU Type=1
+ *     <pre style="white-space: pre;">
+ * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * |       PDU Type(=1)    |         Spare         |
+ * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * |   Spare   |        QoS Flow Identifier        |
+ * +-----+-----+-----+-----+-----+-----+-----+-----+
+ * </pre>
+ *
+ * @author Leo Ma
+ * @since pcap4j 1.7.7
+ */
+public class GtpV1PduSessionContainerExtensionHeader implements GtpV1ExtensionHeader {
+
+  private static final long serialVersionUID = 7361463927403478495L;
+
+  private static final int LENGTH_OFFSET = 0;
+  private static final int LENGTH_SIZE = ByteArrays.BYTE_SIZE_IN_BYTES;
+  private static final int PDU_TYPE_AND_SPARE_OFFSET = LENGTH_OFFSET + LENGTH_SIZE;
+  private static final int PDU_TYPE_AND_SPARE_SIZE = ByteArrays.BYTE_SIZE_IN_BYTES;
+
+  /** DL PDU SESSION INFORMATION (PDU Type 0) */
+  /** Common fields */
+  private static final int PPP_AND_RQI_AND_QFI_OFFSET =
+      PDU_TYPE_AND_SPARE_OFFSET + PDU_TYPE_AND_SPARE_SIZE;
+
+  private static final int PPP_AND_RQI_AND_QFI_SIZE = ByteArrays.BYTE_SIZE_IN_BYTES;
+
+  /** PPI field(presene in case PPP is true) */
+  private static final int PPI_AND_SPARE_OFFSET =
+      PPP_AND_RQI_AND_QFI_OFFSET
+          + PPP_AND_RQI_AND_QFI_SIZE; // should be exist only when PDU Type=0 and PPP flag is true.
+
+  private static final int PPI_AND_SPARE_SIZE = ByteArrays.BYTE_SIZE_IN_BYTES;
+  private static final int PADDING_AND_NEXT_EXTENSION_HEADER_TYPE_OFFSET =
+      PPI_AND_SPARE_OFFSET
+          + PPI_AND_SPARE_SIZE; // should be exist only when PDU Type=0 and PPP flag is true.
+  private static final int PADDING_AND_NEXT_EXTENSION_HEADER_TYPE_SIZE =
+      ByteArrays.INT_SIZE_IN_BYTES;
+
+  /** UL PDU SESSION INFORMATION (PUD Type 1) */
+  private static final int SPARE_AND_QFI_OFFSET =
+      PDU_TYPE_AND_SPARE_OFFSET + PDU_TYPE_AND_SPARE_SIZE;
+
+  private static final int NEXT_EXTENSION_HEADER_TYPE_OFFSET =
+      PPP_AND_RQI_AND_QFI_OFFSET + PPP_AND_RQI_AND_QFI_SIZE;
+  private static final int NEXT_EXTENSION_HEADER_TYPE_SIZE = ByteArrays.BYTE_SIZE_IN_BYTES;
+
+  private static final int GTPV1_PDU_SESSION_CONTAINER_EXTENSION_MIN_LENGTH =
+      NEXT_EXTENSION_HEADER_TYPE_OFFSET + NEXT_EXTENSION_HEADER_TYPE_SIZE;
+  private static final int GTPV1_PDU_SESSION_CONTAINER_EXTENSION_MAX_LENGTH =
+      PADDING_AND_NEXT_EXTENSION_HEADER_TYPE_OFFSET + PADDING_AND_NEXT_EXTENSION_HEADER_TYPE_SIZE;
+
+  private byte length;
+  private final byte pduType;
+
+  /** UL Common */
+  private boolean ppp; // Paging Policy Presence field
+
+  private boolean rqi; // Reflective QoS Indicator field
+  private byte qfi; // Qos Flow Identifier field
+
+  /** UL PPI(presence in case ppp is true) */
+  private byte ppi; // Paging Policy Indicator field
+
+  private int padding;

```

Comment:
I think it's more intuitive (and consistent with other header classes) if `padding` is a byte array.